pages/plant/[id].js

```diff
@@ -0,0 +1,121 @@
+import Image from "next/image";
+import { useRouter } from "next/router";
+import styled from "styled-components";
+import BackLink from "@/components/BackLink";
+
+const StyledSeasonList = styled.ul`
+  list-style: none;
+  padding: 0;
+`;
+
+const IconsContainer = styled.article`
+  display: flex;
+  justify-content: space-around;
+`;
+
+const IconContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+const ImageBorder = styled.article`
+  position: relative;
+  margin-top: 20px;
+  height: 55vw;
+  width: 90vw;
+  border: 0px solid;
+  border-radius: 15px;
+  overflow: hidden;
+  margin: auto;
+`;
+
+export default function PlantDetails({ plants }) {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const plantData = plants ? plants.find((plant) => plant.id === id) : null;
+
+  if (!plantData) return <div>No data available</div>;
+
+  let waterIconSrc = "";
+
+  const waterNeed = plantData.waterNeed;
+  if (waterNeed === "High") {
+    waterIconSrc = "/icons/drop-full.svg";
+  } else if (waterNeed === "Medium") {
+    waterIconSrc = "/icons/drop-half.svg";
+  } else if (waterNeed === "Low") {
+    waterIconSrc = "/icons/drop.svg";
+  }
+
+  let lightIconSrc = "";
+
+  const lightNeed = plantData.lightNeed;
+
+  if (lightNeed === "Full Sun") {
+    lightIconSrc = "/icons/sun.svg";
+  } else if (lightNeed === "Partial Shade") {
+    lightIconSrc = "/icons/sun-half.svg";
+  } else if (lightNeed === "Full Shade") {
+    lightIconSrc = "/icons/sun-full.svg";
+  }
```

Comment:
Here is a suggestion to make it a little bit more DRY. :)
```suggestion
const waterIconSrc = {
  High: "/icons/drop-full.svg",
  Medium: "/icons/drop-half.svg",
  Low: "/icons/drop.svg",
}[plantData.waterNeed] || "";

const lightIconSrc = {
  "Full Sun": "/icons/sun.svg",
  "Partial Shade": "/icons/sun-half.svg",
  "Full Shade": "/icons/sun-full.svg",
}[plantData.lightNeed] || "";
```